### PR TITLE
Add initial ci jobs

### PR DESCRIPTION
**Pull request by cpanato**
- add dependabot
- add initial ci for build/test
- add golangci-lint

### CODE REVIEW

**Comment by cpanato** [comment]
there are some lint issues, that I can fix in a follow-up, after and if we merge this PR, because the jobs added by this PR might not run for this PR

/cc @kaniini @mattmoor

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+      - uses: actions/setup-go@424fc82d43fa5a37540bae62709ddcc23d9520d4 # v2.1.5
+        with:
+          go-version: '1.17.x'
+
+      - name: build
+        run: go build -o apko ./cmd/apko/main.go
+
+  test:
+    name: build
```
Comment (mattmoor) [inline .github/workflows/build.yml:25]: nit: test

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+linters:
```
Comment (mattmoor) [inline .golangci.yml:1]: FYI this is the config we've been using elsewhere:
```
linters:
  enable:
  - asciicheck
  - deadcode
  - depguard
  - errcheck
  - errorlint
  - gofmt
  - goimports
  - gosec
  - gocritic
  - importas
  - prealloc
  - revive
  - misspell
  - stylecheck
  - tparallel
  - unconvert
  - unparam
  - whitespace
output:
  uniq-by-line: false
issues:
  exclude-rules:
  - path: _test\.go
    linters:
    - errcheck
    - gosec
  max-issues-per-linter: 0
  max-same-issues: 0
run:
  issues-exit-code: 1
  timeout: 10m
```

**Review by mattmoor — COMMENTED** [review]
LGTM, defer to @kaniini

File: .golangci.yml (hunk 2 above)
Comment (cpanato) [reply .golangci.yml:1]: hum i will update rekor with that as well :)

Comment (cpanato) [reply .golangci.yml:1]: thanks for the review Matt

**Review by mattmoor — APPROVED** [review]
@kaniini any objections to merging?

**Pull request merged**